raiden/src/utils/ethers.ts

```diff
@@ -107,3 +107,33 @@ export function getEventsStream<T extends any[]>(
 export async function getNetwork(provider: JsonRpcProvider): Promise<Network> {
   return parseNetwork(parseInt(await provider.send('net_version', [])));
 }
+
+/**
+ * Patch JsonRpcProvider.send to try personal_sign first, and fallback to eth_sign if it fails
+ * Call it once on the provider instance
+ *
+ * @param provider  A JsonRpcProvider instance to patch
+ */
+export function patchSignSend(provider: JsonRpcProvider): void {
+  const origSend: (method: string, params: any) => Promise<any> = provider.send;
+  provider.send = async function(method: string, params: any): Promise<any> {
+    if (method === 'eth_sign') {
+      // try 'personal_sign' by default instead of 'eth_sign'
+      return origSend
+        .bind(this)('personal_sign', [params[1], params[0]])
```

Comment:
Also, I realized `bind` isn't appropriate for this, as it will bind the method indefinitelly. Please, replace it with:
```suggestion
        .apply(this, ['personal_sign', [params[1], params[0]]])
```

Reply:
Good catch!